module/finalizer/collection/finalizer_test.go

```diff
@@ -180,15 +179,13 @@ func TestFinalizer(t *testing.T) {
 			assertClusterBlocksIndexedByReferenceHeight(t, db, refBlock.Height, final.ID())
 
 			// block should be passed to provider
-			prov.AssertNumberOfCalls(t, "SubmitLocal", 1)
-			prov.AssertCalled(t, "SubmitLocal", &messages.SubmitCollectionGuarantee{
-				Guarantee: flow.CollectionGuarantee{
-					CollectionID:     block.Payload.Collection.ID(),
-					ReferenceBlockID: refBlock.ID(),
-					ChainID:          block.Header.ChainID,
-					SignerIndices:    block.Header.ParentVoterIndices,
-					Signature:        nil,
-				},
+			prov.AssertNumberOfCalls(t, "SubmitCollectionGuarantee", 1)
```

Comment:
```suggestion
```
This can be replaced by the suggestion on line 151. Assuming we also update to use `collectionmock.NewGuaranteedCollectionPublisher`, the constructor will register this assertion for us.